### ISSUE TITLE
Rebalance Theseus-X

### ISSUE THREAD

**Issue by cswimr** [GalaxyRoblox/Issues-Public#128]
## Final Resolution

Theseus-X module cooldown no longer ticks down while warping or charging warp.

---

## Original Content

In this issue, I'd like for the dev team to evaluate Theseus-X. Quite a few community members have complained about the ship being extremely difficult to reliably finish off, due to its module allowing it to warp around for 30+ minutes without dying. This is because its module allows it to regenerate a large portion of its hull, with no downside.
A possible fix that was suggested to me is making its module cooldown stop counting down while the ship is in warp, but this would require a change to the module system to allow for conditional cooldowns afaik. I'd like to hear other ideas as well, though.
@GalaxyRoblox/dev-advisors @GalaxyRoblox/devs

**Comment by I-Am-Solaris** [comment]
what if you could only use the module like 4 times per spawning of the ship, that would be very easy!

**Comment by I-Am-Solaris** [comment]
you can use modules to put GUI elements on the user's screen as can be seen in Odyssey so it would be trivial to put a uses-remaining counter above the module button

**Comment by cswimr** [comment]
That could work, but would the ship still be fun to use? I feel like disabling the cooldown while in warp is way less invasive than limiting the number of times the module can be used, and it fixes the "warp around forever and never die" issue still.

**Comment by Wingysam** [comment]
Is it really that hard to kill? Looks like it dies pretty often.

<img width="818" alt="image" src="https://github.com/user-attachments/assets/2b69b3fc-dd6b-4ea6-95d5-d25a49dec8f4">

**Comment by cswimr** [comment]
Yeah, it dies pretty often because it's arguably the best dreadnought in the game and is really common. The issue isn't that the ship doesn't die, it's that it is incredibly time consuming to kill it when the pilot is trying to run instead of attempting to dock and despawn like is normal with other ships.

**Comment by Wingysam** [comment]
Okay, what if you disable warping while the module is active?

That may be too harsh though.

**Comment by cswimr** [comment]
I agree with that being too harsh, the standard flow of using the module before you warp out should be maintained. You just shouldn't be able to use it again immediately after exiting warp, that's the issue here imo.

**Comment by Wingysam** [comment]
Dynamic module cooldowns *are* a thing, the module just has to decide its cooldown when the effect ends.

Although the easier solution is to just not return until your state isn't warping/charging. That way the cooldown starts when your warp exits.

**Comment by cswimr** [comment]
When does the Theseus-X module cooldown start? Is it when the module is used or when the effect ends?

**Comment by MRRosdiamond** [comment]
The module needs to have a limit on how often it can be used during a flight.
Either a longer cooldown or a disable it after warping.

If we are looking for more complex mechanic it could be made so that often usage of the module would heavily affect the % it heals.
Using it for the first time in a while will result in a great heal while spamming the module will bare minimal results.

**Comment by cswimr** [comment]
Wait, the simpler solution would be to just increase its cooldown substancially. I don't know why I didn't think of that lol

**Comment by Wingysam** [comment]
All module cooldowns start after their effect function returns.

**Comment by cswimr** [comment]
Ah yeah right, checking the ship's state makes the most sense I think and doesn't nerf the ship's intended usage.

**Comment by PataNautic** [comment]
> Dynamic module cooldowns _are_ a thing, the module just has to decide its cooldown when the effect ends.
> 
> Although the easier solution is to just not return until your state isn't warping/charging. That way the cooldown starts when your warp exits.

Yeah, this seems like the best method

**Comment by cswimr** [comment]
Alright, so this is the plan currently:
- Extend the duration of the Theseus-X's module from 10 seconds to 15 seconds. The time between healing ticks will be slightly increased (12 t/s to 8 t/s), but the overall amount of healing will remain the same (3,502 hull in total over 120 ticks, with 29.19 hull per tick).
- If the Theseus-X begins charging warp or warping during the module's duration, the module's cooldown does not start until the ship exits warp.

Does anyone have any problems with this solution?

**Comment by Wingysam** [comment]
Consider pro-rating the cooldown if the warp is cancelled. If the state changes to normal without first changing to warping, subtract however many seconds they spent charging warp from the cooldown timer (min 0 to prevent negative cooldowns).

**Comment by cswimr** [comment]
After some further discussion with the team, this is what we've decided to do.
We will be playtesting these changes in Dev before they are added to Main. This issue is now blocked by the cooldown change.
## Theseus-X Rebalance
- Module
  - Hull per Tick: 29.19 --> 37.5
  - Duration: 10 --> 20
  - Ticks: 120 *(unchanged)*
  - Total health gained: 3,502 --> 4,500
  - Make the cooldown not tick down in warp
- 2k shield nerf
  - applies to all Theseus variants above whichever one buffs health over baseline
- Plasma Defense System (X and K's AA turrets) range decreased from 5,500 to 4,500
- Remove passive hull regen

**Comment by PataNautic** [comment]
Seems more fair, but yea needs testing

**Comment by I-Am-Solaris** [comment]
Moved to "In Progress" since it's a part of the Limited rebalance that's being worked on. Still needs ability to prevent module CD from reducing in warp.

**Comment by PataNautic** [comment]
Did this never get pushed 😭 @I-Am-Solaris

**Comment by SeanMorabitoYT** [comment]
LoL
